File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>
#include <bitset>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    
    int count,zeroCount;
    
    while (1){
        string temp;
        if(temp.size()==1)
            break;
        for(char a: s){
            if(a=='0'){
                zeroCount++;
            } else{
                temp+=a;
            }
        }
        int len = temp.size();
        temp = bitset<64>(len).to_string();
        count+=1;
    }
    answer.push_back(count,zeroCount);
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    answer = []
    zeroCount=0
    count=0
    temp=s
    while(len(temp)!=1):
        result=""
        for i in temp:
            if (i == '0'):
                zeroCount+=1
            else:
                result+=i
        b= bin(len(result))
        temp=str(b[2:])
        count+=1

    answer.append(count)
    answer.append(zeroCount)
    return answer